SensusService/SensusServiceHelper.cs

```diff
@@ -806,98 +802,64 @@ public void StartAsync(Action callback)
+            var scriptsWithSameParent = _scriptsToRun.Where(s => s.SharesParentScriptWith(script)).ToArray();
 
-                List<Script> scriptsWithSameParent = _scriptsToRun.Where(s => s.SharesParentScriptWith(script)).ToList();
+            if (scriptsWithSameParent.Any() && runMode == RunMode.SingleKeepOldest)
+            {
+                return;
+            }
 
-                if (scriptsWithSameParent.Count > 0)
+            if (scriptsWithSameParent.Any() && runMode == RunMode.SingleUpdate)
+            {
+                foreach (var scriptWithSameParent in scriptsWithSameParent)
                 {
-                    if (runMode == RunMode.SingleKeepOldest)
-                        add = false;
-                    else if (runMode == RunMode.SingleUpdate)
-                        foreach (Script scriptWithSameParent in scriptsWithSameParent)
-                            _scriptsToRun.Remove(scriptWithSameParent);
+                    _scriptsToRun.Remove(scriptWithSameParent);
                 }
+            }
 
-                if (add)
-                {
-                    _scriptsToRun.Insert(0, script);
-                    IssuePendingSurveysNotificationAsync(true, true);
-                }
-            });
+            _scriptsToRun.Insert(0, script);
+            IssuePendingSurveysNotificationAsync(true, true);
         }
 
         public void RemoveScriptToRun(Script script)
         {
-            RunOnMainThread(() =>
-            {
-                if (_scriptsToRun.Remove(script))
-                    IssuePendingSurveysNotificationAsync(false, false);
-            });
+            RemoveScripts(true, script);
         }
 
         public void RemoveScriptsToRun(ScriptRunner runner)
         {
-            RunOnMainThread(() =>
-            {
-                bool removed = false;
-
-                foreach (Script scriptFromRunner in _scriptsToRun.Where(script => ReferenceEquals(script.Runner, runner)).ToList())
-                    if (_scriptsToRun.Remove(scriptFromRunner))
-                        removed = true;
-
-                if (removed)
-                    IssuePendingSurveysNotificationAsync(false, false);
-            });
+            RemoveScripts(true, _scriptsToRun.Where(script => script.Runner == runner).ToArray());
         }
 
         public void RemoveOldScripts(bool issueNotification)
         {
-            RunOnMainThread(() =>
-            {
-                bool removed = false;
-
-                foreach (Script script in _scriptsToRun.ToList())
-                    if (script.Runner.MaximumAgeMinutes.HasValue && script.Age.TotalMinutes >= script.Runner.MaximumAgeMinutes && _scriptsToRun.Remove(script))
-                        removed = true;
-
-                if (removed && issueNotification)
-                    IssuePendingSurveysNotificationAsync(false, false);
-            });
+            RemoveScripts(true, _scriptsToRun.Where(Expired).ToArray());
```

Comment:
Don't pass `true` since the caller is telling you whether to issue a notification. Pass `issueNotification`.